### DISCUSSION TITLE
Support a special symbol for generating IS NULL in SQL queries when `ignoreUndefinedInQuery` is enabled

### DISCUSSION THREAD

**Discussion by sevenryze** [mikro-orm/mikro-orm#6262] [Ideas]
Currently, when `ignoreUndefinedInQuery` is set to true, MikroORM ignores undefined in query conditions but respects null for generating IS NULL in SQL. However, many JavaScript practices and widely-used linter configurations (e.g., Unicron) discourage the use of the null keyword.

To align with such practices, it would be great if MikroORM could support a special symbol (e.g., Symbol('IS_NULL')) that explicitly indicates the need to generate IS NULL in SQL queries.

This would allow developers to avoid using null while still leveraging the ability to query for NULL values in the database.

Proposed Example:
```ts
import { IS_NULL } from '@mikro-orm/core';

const result = await em.find(EntityName, {
  someField: IS_NULL, // Generates `WHERE some_field IS NULL`
});
```
This approach improves readability, adheres to modern JavaScript conventions, and maintains clear semantics for IS NULL queries.

Would it be feasible to add support for such a feature?

**Comment by B4nan** [comment]
I can't help myself, but `someField: null` is a superb solution when it comes to readability, using some symbols will never be as readable as that. Blindly obeying linter is something I never liked/understood.

It's surely doable, and I wouldn't mind accepting a PR for this I guess, but I really don't like it 🙃

**Comment by sevenryze** [comment]
Sure，I got your point. I’ll think about it and get back to you if I have
plan to implement this PR.
